src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs

```diff
@@ -167,11 +167,22 @@ private static ObjectType SetBicepResourceProperties(ObjectType objectType, Reso
                 // we can refer to a resource at any scope if it is an existing resource not being deployed by this file
                 var scopeReference = LanguageConstants.CreateResourceScopeReference(validParentScopes);
                 properties = properties.SetItem(LanguageConstants.ResourceScopePropertyName, new TypeProperty(LanguageConstants.ResourceScopePropertyName, scopeReference, scopePropertyFlags));
+
+                if (properties.TryGetValue(LanguageConstants.ResourceLocationPropertyName, out var locationTypeProperty))
+                {
+                    // An existing resource's location is not a deployment-time constant, because it requires runtime evaluation get the value.
+                    properties = properties.SetItem(LanguageConstants.ResourceLocationPropertyName, new TypeProperty(LanguageConstants.ResourceLocationPropertyName, locationTypeProperty.TypeReference, locationTypeProperty.Flags & ~TypePropertyFlags.DeployTimeConstant));
+                }
             }
             else
             {
                 // TODO: remove 'dependsOn' from the type library
                 properties = properties.SetItem(LanguageConstants.ResourceDependsOnPropertyName, new TypeProperty(LanguageConstants.ResourceDependsOnPropertyName, LanguageConstants.ResourceOrResourceCollectionRefArray, TypePropertyFlags.WriteOnly));
+
+                if (properties.TryGetValue(LanguageConstants.ResourceLocationPropertyName, out var locationTypeProperty))
+                {
+                    properties = properties.SetItem(LanguageConstants.ResourceLocationPropertyName, new TypeProperty(LanguageConstants.ResourceLocationPropertyName, locationTypeProperty.TypeReference, locationTypeProperty.Flags | TypePropertyFlags.DeployTimeConstant));
```

Comment:
Should I also update the type generator to tag `location` as `DeployTimeConstant`?

Reply:
Good pt

---
In reply to: [613672905](https://github.com/Azure/bicep/pull/2261#discussion_r613672905) [](ancestors = 613672905)

Reply:
I'll create a PR later.